### PR TITLE
Upgrade to Lodash4

### PR DESCRIPTION
**Pull request by nathanstitt**
*Note: *  I'm experiencing a spec failure, which I think is because the spec itself is invalid.

This has been resolved.  I needed to switch `sortedIndex` out for `sortedIndexBy`
~~~The failure occurs in the spec for testing ordering is saved.~~~

~~~However, `sweet` is set to false on line 350, so that model is removed from the sub-collection.  Accordingly I don't believe it should be present when the collection is sorted.~~~

~~~Maybe I'm mis-reading the code though and this is a side-effect of the upgrade?~~~

### CODE REVIEW

File: ampersand-filtered-subcollection.js [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
-var isArray = require('lodash.isarray');
-var isEqual = require('lodash.isequal');
-var keys = require('lodash.keys');
-var reduce = require('lodash.reduce');
-var sortBy = require('lodash.sortby');
-var sortedIndex = require('lodash.sortedindex');
-var union = require('lodash.union');
+var includes = require('lodash/includes');
+var difference = require('lodash/difference');
+var forEach = require('lodash/foreach');
```
Comment (pgilad) [inline ampersand-filtered-subcollection.js]: some OSes are case sensitive so you need to import `lodash/forEach`

File: ampersand-filtered-subcollection.js [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
+var every = require('lodash/every');
+var assign = require('lodash/assign');
+var isArray = require('lodash/isarray');
```
Comment (pgilad) [inline ampersand-filtered-subcollection.js]: `isArray`

File: ampersand-filtered-subcollection.js [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
+var isEqual = require('lodash/isequal');
```
Comment (pgilad) [inline ampersand-filtered-subcollection.js]: `isEqual`

File: ampersand-filtered-subcollection.js [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
+var keys = require('lodash/keys');
+var reduce = require('lodash/reduce');
+var sortBy = require('lodash/sortBy');
+var sortedIndex = require('lodash/sortedindex');
```
Comment (pgilad) [inline ampersand-filtered-subcollection.js]: `sortedIndex`

**Comment by nathanstitt** [comment]
@pgilad Thanks for pointing out the camelCase issue. I've corrected that and tested on an Ubuntu VM to double-check.

This keeps API compatibility so would not strictly require a server bump, however it'd probably still be wise to do so IMO.

**Comment by nathanstitt** [comment]
Oh and sauce labs appears to be having some issues, the specs all pass locally on both OS X and Ubuntu.

**Comment by lukekarrys** [comment]
Closing this since #31 was merged. Thanks for your hard work on this!

**Pull request closed without merging**